File: code/generate_strategies/RA(method)/python/function_shutil_copytree.py
import shutil
import os


class DirectoryHandler:
    def handle_directory_operations(self, src_dir: str, dest_dir: str, operation: str = "copy") -> str:
        
        if operation == "copy":
            if os.path.exists(dest_dir):
                raise FileExistsError(f"The destination directory '{dest_dir}' already exists.")
            shutil.copytree(src_dir, dest_dir)
            return f"Directory copied from {src_dir} to {dest_dir}."
        else:
            raise ValueError("Unsupported operation. Only 'copy' is supported.")

File: manual/code/generate_strategies/Direct/cpp/function_shutil_copytree.cpp

#include <iostream>
#include <string>
#include <boost/filesystem.hpp>
#include <fstream>
class DirectoryHandler {
public:
    std::string handle_directory_operations(std::string src_dir, std::string dest_dir, std::string operation = "copy") {
        using namespace boost::filesystem;
        if (operation == "copy") {
            if (exists(dest_dir)) {
                throw std::runtime_error("The destination directory '" + dest_dir + "' already exists.");
            }
            boost::filesystem::copy(src_dir, dest_dir);
            return "Directory copied from " + src_dir + " to " + dest_dir + ".";
        } else {
            throw std::runtime_error("Unsupported operation. Only 'copy' is supported.");
        }
    }
};